src/core.js

```diff
@@ -212,7 +212,7 @@ jQuery.extend( {
 	noop: function() {},
 
 	isFunction: function( obj ) {
-		return jQuery.type( obj ) === "function";
+		return obj instanceof Function && typeof obj === "function";
```

Comment:
Is that really make sense to regard it as an exception? I wonder other exceptions could appear.
The documentation says:
> Determine if the argument passed is a JavaScript function object.

Is that behavior appropriate for the description? It is no longer valid to call the implementation as `jQuery.isFunction` if it is too far from the existing one. It is just confusing.